keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -650,450 +650,484 @@ where SCL is the apiserver's concurrency limit and ACS(l) is the
-way, which corresponds more directly to max-min fairness.  We have a
-collection of FIFO queues of packets to be transmitted, one packet at
-a time, over one link that goes at a fixed rate of `mu_single` (in
-units of bits/second).  We imagine a virtual world called concurrent
-service (CS), in which transmission is proceeding for all the
-non-empty queues concurrently; queue `i` is transmitted at rate
-`mu(i,t)` at time `t`.  At any given `t` the allocations are given by
+- Each request `(i,j)` comes tagged with `width(i,j)`, the integer
+  number of seats it occupies.  This is typically 1, but can be up to
+  a configured limit `A` (typically a double-digit number) in some cases.
 
-```
-mu(i,t) = min(rho(i,t), mu_fair(t))
-```
+- At any given moment the sum of the widths of the executing requests
+  should be as high as it can be without exceeding `C`, but if that
+  would not allow any requests to be executing then instead there
+  should be one executing.
 
-where:
-- `i` identifies a queue,
-- `rho(i,t)` is the rate requested by queue `i` at time `t` and is
-  defined to be the product of `mu_single` and the number of packets
-  of that queue that are not fully sent in the virtual world at time
-  `t` (those for which `t_arrive(i,j) <= t` and whose transmission
-  completes strictly after `t`), and
-- `mu_fair(t)` is the smallest non-negative quantity that solves the equation
-  ```
-  min(mu_single, Sum[over i] rho(i,t)) = Sum[over i] min(rho(i,t), mu_fair(t))
-  ```
+- A request can be ejected before it has been dispatched.  This means
+  to remove the request and never execute it.
 
-That implies that `mu(i,t)` is zero for each empty queue and
-`mu_single / (number of non-empty queues at time t)` for each
-non-empty queue, where emptiness is judged in the virtual world.
+- A request does _not_ come with an indication of how long it will
+  take to execute.  That is only known when the execution finishes.
 
-In the virtual world bits are transmitted from queue `i` at rate
-`mu(i,t)`, for every non-empty queue `i`.  Each time when a queue
-transitions from empty to non-empty is when bits start being sent from
-that queue in the virtual world; the solution `mu_fair(t)` gets
-adjusted at this time, among others.  In this virtual world a queue's
-packets are divided into three subsets: those that have been
-completely sent, those that are in the process of being sent, and
-those that have not yet started being sent.  That number being sent is
-1 unless it is 0 and the queue has no unsent packets.  This virtual
-world uses the same clock as the real world.  Whenever a packet
-finishes being sent in the real world, the next packet to be
-transmitted is the one that is unsent in the real world and will
-finish being sent soonest in the virtual world.  If there is a tie
-among several, we pick the one whose queue is next in round-robin
-order (following the queue last picked).
+- A request _does_ come tagged with some extra time `tarry(i,j)` to
```

Comment:
I think "additional latency" is pretty much what "tarry" is about.
- https://www.merriam-webster.com/dictionary/tarry
- https://www.dictionary.com/browse/tarry
- https://www.vocabulary.com/dictionary/tarry

I disfavor "padding" because it does not so clearly indicate that we are talking about time.

Reply:
In the force-push to 1a58c3eb I replaced `tarry` with `extra_latency`.